plugins/org.jboss.reddeer.eclipse/src/org/jboss/reddeer/eclipse/wst/xml/ui/tabletree/XMLSourcePage.java

```diff
@@ -62,52 +54,12 @@ protected XMLSourcePage(ITextEditor editor) {
 	 * @return the string
 	 */
 	public String evaluateXPath(String xPathExpression){
-		XPath xPath =  XPathFactory.newInstance().newXPath();
-
 		try {
-			Document doc = getDocument();
-			xPath.setNamespaceContext(new UniversalNamespaceResolver(doc));
-			return xPath.compile(xPathExpression).evaluate(doc);
+			XPathEvaluator xPathEvaluator = new XPathEvaluator(new ByteArrayInputStream(getText().getBytes()));
```

Comment:
You have getAssociatedFile method in Editor. What about overriding the method in this class and call evaluation of xpath in following manner xmlSourcePage.getAssociatedFile.xpath(myExpression)? Because now for some editor you have it done this way. But in this particular its different - ambiguous